lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*****************************************************************************
+  Source      	SetMessageManagerHandler.cpp
+  Version     	0.1
+  Date       	2020/08/08
+  Product     	SessionD
+  Subsystem   	5G Landing object in SessionD
+  Author/Editor Sanjay Kumar Ojha
+  Description 	Acts as 5G Landing object in SessionD & start 5G related flow
+*****************************************************************************/
+#include <chrono>
+#include <thread>
+
+#include <google/protobuf/util/time_util.h>
+
+#include "SetMessageManagerHandler.h"
+#include "magma_logging.h"
+#include "GrpcMagmaUtils.h"
+
+using grpc::Status;
+
+namespace magma
+{
+
+/**
+ * SetMessageManagerHandler processes gRPC requests for the sessionD
+ * This composites the earlier LocalSessionManagerHandlerImpl and uses the
+ * exiting functionalities.
+ */
+
+/*Constructor*/
+SetMessageManagerHandler::SetMessageManagerHandler(
+    std::shared_ptr<SessionStateEnforcer> m5genforcer, SessionStore& session_store)
+    : session_store_(session_store),
+      m5g_enforcer_(m5genforcer) {}
+
+/*Building session config with required parameters*/  //TODO on SubscriberID how to assigne or manage
+SessionConfig SetMessageManagerHandler::m5g_build_session_config(

```

Comment:
I don't think we need this, we can use the same `SessionConfig` constructor we use in the 4G case.

Reply:
Code wise the assignment are exactly similar. Do you see any advantage of defining constructor here? Let me know accordingly we can adapt.

Reply:
Existing constructor can't be used here. As discussed, added respective TODO and in future will add constructor w.r.t. 5G